CDTDatastore/HTTP/CDTURLSessionTask.m

```diff
@@ -203,15 +202,19 @@ - (void) completedThread:(NSThread *)thread {
         [self.inProgressTask resume];
     } else {
         if( self.requestError){
-            MYOnThread(thread, ^{
-                [self.delegate requestDidError:self.requestError];
-            });
+            [self.delegate performSelector:@selector(requestDidError:)
+                                  onThread:thread
+                                withObject:self.requestError
+                             waitUntilDone:NO];
         } else {
-            MYOnThread(thread, ^{
-                [self.delegate receivedResponse:self.response];
-                [self.delegate receivedData:self.requestData];
-            });
-            
+            [self.delegate performSelector:@selector(receivedResponse:)
```

Comment:
The behaviour of this is now changed by waiting on the current thread for the `receivedResponse:` to complete and then calling `receivedData` whereas previously they were both called without waiting on the current thread. I think perhaps the method calls:

> [self.delegate receivedResponse:self.response];
> [self.delegate receivedData:self.requestData];

should be put in a separate wrapper method and then called with a single call to `performSelector` that calls the wrapper method on the separate thread and doesn't wait until done.